core/src/impl/Kokkos_Atomic_Generic.hpp

```diff
@@ -165,6 +202,7 @@ KOKKOS_INLINE_FUNCTION T atomic_fetch_oper(
   } oldval, assume, newval;
 
   oldval.t = *dest;
+  if (check_early_exit(Oper{}, oldval.t, val)) return oldval.t;
```

Comment:
Needs to be in the while loop

Reply:
Yeah, @crtrott I'm not sure why the early exit check needs to be in the while loop. Probably the fewer branches the better in the CAS loop itself

Reply:
https://github.com/desul/desul/pull/13 has our discussion of this topic.  It may be easier to have it over there.